channel/consensus_channel/leader_channel_test.go

```diff
@@ -125,17 +126,120 @@ func TestLeaderChannel(t *testing.T) {
 	thirdChannel := types.Destination{3}
 	p2 := p
 	p2.target = thirdChannel
-	_, err = channel.Propose(p2, testdata.Actors.Alice.PrivateKey)
+	g2 := p2.Guarantee
+	secondSigned, err := channel.Propose(p2, testdata.Actors.Alice.PrivateKey)
 	if err != nil {
 		t.Fatalf("failed to add another proposal: %v", err)
 	}
+	if secondSigned.Proposal.(Add).turnNum != 2 {
+		t.Fatalf("incorrect proposal generated")
+	}
 
-	success, _ = channel.IsProposed(p2.Guarantee)
+	success, _ = channel.IsProposed(g2)
 	if !success {
 		t.Fatal("incorrect latest proposed vars")
 	}
 	if channel.ConsensusTurnNum() != 0 {
 		t.Fatal("consensus incorrectly updated")
 	}
+	if channel.Includes(g2) {
+		t.Fatal("consensus incorrectly updated")
+	}
+
+	latest, _ := channel.latestProposedVars()
+	counterSig2, _ := latest.asState(fp()).Sign(testdata.Actors.Bob.PrivateKey)
+
+	p3 := p
+	p3.target = types.Destination{4}
+	g3 := p3.Guarantee
+	thirdSigned, _ := channel.Propose(p3, testdata.Actors.Alice.PrivateKey)
+
+	p2Returned := SignedProposal{
+		Proposal:  secondSigned.Proposal,
+		Signature: counterSig2,
+	}
+
+	// A counter signature is received on a proposal (but not the latest proposal)
+	err = channel.UpdateConsensus(p2Returned)
+	if err != nil {
+		t.Fatalf("Unable to update consensus: %v", err)
+	}
+
+	if channel.ConsensusTurnNum() != 2 {
+		t.Fatalf("consensus turn num not updated")
+	}
+	if !channel.Includes(g2) {
+		t.Fatal("consensus incorrectly updated")
+	}
+	if channel.Includes(g3) {
+		t.Fatal("consensus incorrectly updated")
+	}
+
+	err = channel.UpdateConsensus(p2Returned)
+	if err != nil {
+		t.Fatalf("Unable to update consensus: %v", err)
+	}
```

Comment:
I think these lines are duplicated from L163-L166

Reply:
@andrewgordstewart do you agree?

Reply:
yes, sorry!

addressing this in https://github.com/statechannels/go-nitro/issues/417